### PR TITLE
[Feature]Extend test coverage across multiple model architectures using the Scoring API

### PR DESCRIPTION
**Pull request by Lj1ang**
## Motivation

This PR extends test coverage for the Scoring API to validate correctness across multiple model architectures. Previously, the Scoring API tests primarily focused on a single model architecture. This enhancement ensures the API works correctly across different model families and architectures, including:

- **CausalLM models**: Llama (instruct and base variants), Qwen
- **SequenceClassification models**: Qwen3ForSequenceClassification

This comprehensive test coverage helps catch architecture-specific bugs and ensures consistent behavior across different model types, which is critical for production reliability.

ref: Extend test coverage across multiple model architectures using the Scoring API in  https://github.com/sgl-project/sglang/issues/15344

## Modifications

- **Added new test file**: `test/registered/core/test_score_api_multi_arch.py` (516 lines)
  - Comprehensive integration tests for Scoring API across multiple architectures
  - Parameterized test suite that automatically generates tests for each model architecture
  - Reference implementations using HuggingFace for accuracy validation

**Test Coverage:**
- Basic scoring tests for each architecture (Llama instruct, Llama base, Qwen)
- Cross-architecture consistency validation
- Edge cases: empty queries, single items, single/many label tokens
- Error handling: invalid token IDs
- Advanced features: `item_first` parameter, softmax normalization options, large batch processing
- Parameterized tests for all supported model architectures

**Models Tested:**
- `causallm_llama`: Llama-3.2-1B-Instruct
- `causallm_llama_base`: Llama-3.2-1B (base model)
- `causallm_qwen`: Qwen2.5-1.5B-Instruct
- `sequence_classification_qwen3`: Qwen3-Reranker-0.6B

## Accuracy Tests

All tests compare SGLang Scoring API outputs with HuggingFace reference implementations to ensure numerical accuracy:

- **CausalLM models**: Validates token logits/probabilities match HuggingFace outputs (tolerance: 0.01)
- **SequenceClassification models**: Validates classification scores match reference implementations
- Tests cover both softmax-normalized probabilities and raw logits
- Large batch processing validation
- Cross-architecture consistency checks

**Test Results:**
- All 15+ tests pass successfully
- Tests registered in CI with estimated time: 600s
- CI suite: `stage-b-test-large-1-gpu`

## Benchmarking and Profiling

N/A - This PR only adds test coverage and does not modify inference code or impact performance.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: test/registered/core/test_score_api_multi_arch.py [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+                    tolerance,
+                    msg=f"Scores differ by {diff:.6f} ({case_name}): "
+                    f"HF={hf_score:.6f}, SGLang={sglang_score:.6f}",
+                )
+
+    # ==================== CausalLM Architecture Tests ====================
+
+    def test_causallm_llama_basic_scoring(self):
+        """Test basic scoring with Llama CausalLM model."""
+        self._setup_model(TEST_MODELS["causallm_llama"], "causallm")
```
Comment (sundar24295s) [inline test/registered/core/test_score_api_multi_arch.py:203]: - Every test method calls _setup_model() in its body, and tearDown() shuts the engine down after each. 
- Instead, we should load the test model once per Test suite and run the test methods.

File: test/registered/core/test_score_api_multi_arch.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+register_cuda_ci(est_time=600, suite="stage-b-test-large-1-gpu")
+# Test models for different architectures
+# Using smaller models for faster CI execution
+TEST_MODELS = {
+    # CausalLM models
+    "causallm_llama": DEFAULT_SMALL_MODEL_NAME_FOR_TEST,  # Llama-3.2-1B-Instruct
+    "causallm_llama_base": DEFAULT_SMALL_MODEL_NAME_FOR_TEST_BASE,  # Llama-3.2-1B
+    "causallm_qwen": DEFAULT_SMALL_MODEL_NAME_FOR_TEST_QWEN,  # Qwen2.5-1.5B-Instruct
+    # SequenceClassification models
+    "sequence_classification_qwen3": DEFAULT_SMALL_MODEL_NAME_FOR_TEST_SCORE,  # Qwen3-Reranker-0.6B (Qwen3ForSequenceClassification)
```
Comment (sundar24295s) [inline test/registered/core/test_score_api_multi_arch.py:42]: - We don't have support for Sequence Classification models yet in the Score API.

File: test/registered/core/test_score_api_multi_arch.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+        )
+
+        self.assertEqual(len(sglang_scores), len(items))
+        for score_list in sglang_scores:
+            self.assertEqual(len(score_list), len(label_token_ids))
+            self.assertAlmostEqual(sum(score_list), 1.0, places=4)
+
+    # ==================== Cross-Architecture Consistency Tests ====================
+
+    def test_consistency_across_architectures(self):
```
Comment (sundar24295s) [inline test/registered/core/test_score_api_multi_arch.py:332]: - This test is not needed. 
- Effectively, It asserts both produce the same number of scores and both sum to 1. That's true by construction when apply_softmax=True

**Review by sundar24295s — CHANGES_REQUESTED** [review]
- I think we can hold off on merging for now — most of the CausalLM coverage here overlaps with what's already in test_score_api.
- Would recommened to extend test_score_api.py to parameterize across CausalLM variants (Llama, Qwen), and add sequence classification tests once that support lands.

File: test/registered/core/test_score_api_multi_arch.py (hunk 3 above)
Comment (Lj1ang) [reply test/registered/core/test_score_api_multi_arch.py:332]: Sorry for my oversight, that's true.

**Comment by Lj1ang** [comment]
> * I think we can hold off on merging for now — most of the CausalLM coverage here overlaps with what's already in test_score_api.
> * Would recommened to extend test_score_api.py to parameterize across CausalLM variants (Llama, Qwen), and add sequence classification tests once that support lands.

I strongly agree with these.

**Comment by hnyls2002** [comment]
@Lj1ang Thanks for the contribution. Per the review feedback from @sundar24295s, the plan was to fold this coverage into the existing test_score_api.py (now at test/registered/prefill_only/test_score_api.py) rather than add a parallel test file, and you agreed with that direction. Since there has been no follow-up for several months, I am closing this to keep the queue tidy. Please reopen or send a new PR that extends test_score_api.py directly if you would like to pick this back up.
